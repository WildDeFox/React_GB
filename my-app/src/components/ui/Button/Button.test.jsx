import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'

import Button from './Button'

describe('Button', () => {
  it('render component Button', () => {
    render(<Button>Change name</Button>)
  })

  it('render with snapshot', () => {
    const {asFragment} = render(<Button>Change name</Button>)
    expect(asFragment()).toMatchSnapshot()
  })

  it('render component with text name button', () => {
    render(<Button>Change name</Button>) 
    expect(screen.getByText(/Change name/)).toBeInTheDocument()
  })

  it('render multiply component', () => {
    render(
      <>
        <Button>Click</Button>
        <Button>Delete</Button>
      </>
    )
    expect(screen.getAllByRole('button').length).toBe(2)
  })

  it('button is disable', () => {
    render(<Button disabled>Delete</Button>)
    expect(screen.getByText('Delete')).toBeDisabled()
  })

  it('button have style color green', () => {
    render(<Button>Click</Button>)
    expect(screen.getByText('Click')).toHaveStyle({color: 'green'})
  })

  it('button click with userEvent', async () => {
    const mockHandler = jest.fn()
    render(<Button click={mockHandler}>Click</Button>)
    await userEvent.click(screen.getByText('Click'))
    expect(mockHandler).toHaveBeenCalledTimes(1)
  })

  it('text example', async () => {
    const onChange = jest.fn()
    render(<input type="checkbox" onChange={onChange} />)

    const checkbox = screen.getByRole('checkbox')
    await userEvent.dblClick(checkbox)
    expect(onChange).toHaveBeenCalledTimes(2)
    expect(checkbox).not.toBeChecked()
  })
})