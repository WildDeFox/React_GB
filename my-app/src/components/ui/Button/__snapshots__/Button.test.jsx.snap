// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render with snapshot 1`] = `
<DocumentFragment>
  <button
    style="color: green;"
  >
    Change name
  </button>
</DocumentFragment>
`;
